RecFCCeeCalorimeter/src/components/PhotonIDTool.cpp

```diff
@@ -0,0 +1,361 @@
+      auto it = std::find(shapeParameters.begin(), shapeParameters.end(), feature);
+      if (it != shapeParameters.end())
+      {
+        int position = std::distance(shapeParameters.begin(), it);
+        m_inputPositionsInShapeParameters.push_back(position);
+        info() << "Feature " << feature << " found in position " << position << " of shapeParameters" << endmsg;
+      }
+      else
+      {
+        // at least one of the inputs of the MVA was not found in the shapeParameters
+        // so we can stop checking the others
+        m_inputPositionsInShapeParameters.clear();
+        error() << "Feature " << feature << " not found, aborting..." << endmsg;
+        return StatusCode::FAILURE;
+      }
+    }
+  }
+    
+  // append the MVA score to the output shape parameters
+  shapeParameters.push_back("photonIDscore");
+  m_outShapeParameterHandle.put(shapeParameters);
+
+  info() << "Initialized the photonID MVA tool" << endmsg;
+  return StatusCode::SUCCESS;
+}
+
+StatusCode PhotonIDTool::execute([[maybe_unused]] const EventContext &evtCtx) const
+{
+  verbose() << "-------------------------------------------" << endmsg;
+
+  // Get the input collection with clusters
+  const edm4hep::ClusterCollection *inClusters = m_inClusters.get();
+
+  // Initialize output clusters
+  edm4hep::ClusterCollection *outClusters = initializeOutputClusters(inClusters);
+  if (!outClusters)
+  {
+    error() << "Something went wrong in initialization of the output cluster collection, exiting!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (inClusters->size() != outClusters->size())
+  {
+    error() << "Sizes of input and output cluster collections does not match, exiting!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // Run inference
+  {
+    StatusCode sc = applyMVAtoClusters(inClusters, outClusters);
+    if (sc.isFailure())
+    {
+      return sc;
+    }
+  }
+
+  return StatusCode::SUCCESS;
+}
+
+StatusCode PhotonIDTool::finalize()
+{
+  if (m_ortSession)
+    delete m_ortSession;
+  if (m_ortEnv)
+    delete m_ortEnv;
+
+  return Gaudi::Algorithm::finalize();
+}
+
+edm4hep::ClusterCollection *PhotonIDTool::initializeOutputClusters(
+    const edm4hep::ClusterCollection *inClusters) const
+{
+  edm4hep::ClusterCollection *outClusters = m_outClusters.createAndPut();
+
+  for (auto const &inCluster : *inClusters)
+  {
+    auto outCluster = inCluster.clone();
+    outClusters->push_back(outCluster);
+  }
+
+  return outClusters;
+}
+
+StatusCode PhotonIDTool::readMVAFiles(const std::string& mvaInputsFileName,
+                                      const std::string& mvaModelFileName)
+{
+  // 1. read the file with the list of input features
+  // Open the JSON file
+  std::ifstream file(mvaInputsFileName);
+  if (!file.is_open()) {
+    error() << "Error opening file: " << mvaInputsFileName << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // Parse the JSON file
+  json j;
+  file >> j;
```

Comment:
What happens if the file contains non valid JSON?

Reply:
failing miserably I guess, need to probably put this in some try .. catch block. On my todo for next week

Reply:
fixed